tests/lms/views/error_test.py

```diff
@@ -1,106 +1,89 @@
+        exc = HTTPNotImplemented()
 
-        error_views = error.ErrorViews(exc, pyramid_request)
-        error_views.httperror()
+        ErrorViews(exc, pyramid_request).http_error()
 
-        assert pyramid_request.response.status_int == exc.status_int
+        assert pyramid_request.response.status_int == 501
 
-    def test_it_sets_correct_message_for_http_server_error(self, pyramid_request):
-        exc = httpexceptions.HTTPServerError('test server error msg', status_code=500)
+    def test_http_error_shows_the_exception_message_to_the_user(self, pyramid_request):
+        exc = HTTPNotImplemented("This is the error message")
 
-        error_views = error.ErrorViews(exc, pyramid_request)
-        resp = error_views.httperror()
+        result = ErrorViews(exc, pyramid_request).http_error()
 
-        assert resp['message'] == 'test server error msg'
+        assert result["message"] == "This is the error message"
 
-    def test_it_sets_correct_request_status_int_for_http_server_error(self, pyramid_request):
-        exc = httpexceptions.HTTPServerError('test server error msg', status_code=500)
+    def test_lti_launch_error_reports_exception_to_sentry(
+        self, pyramid_request, sentry_sdk
+    ):
+        exc = LTILaunchError("the_message")
 
-        error_views = error.ErrorViews(exc, pyramid_request)
-        error_views.httperror()
+        ErrorViews(exc, pyramid_request).lti_launch_error()
 
-        assert pyramid_request.response.status_int == exc.status_int
+        sentry_sdk.capture_exception.assert_called_once_with(exc)
 
-    def test_it_sets_correct_message_for_non_http_error(self, pyramid_request):
-        exc = Exception()
+    def test_lti_launch_error_sets_response_status(self, pyramid_request):
+        exc = LTILaunchError("the_message")
 
-        error_views = error.ErrorViews(exc, pyramid_request)
-        resp = error_views.error()
+        ErrorViews(exc, pyramid_request).lti_launch_error()
 
-        assert resp['message'] == 'Sorry, but something went wrong. The issue has been reported and we\'ll try to fix it.'
-
-    def test_it_sets_correct_status_int_for_non_http_error(self, pyramid_request):
-        exc = Exception()
-
-        error_views = error.ErrorViews(exc, pyramid_request)
-        resp = error_views.error()
-
-        assert pyramid_request.response.status_int == 500
-
-    def test_it_logs_non_http_error_in_sentry(self, pyramid_request):
-        exc = Exception()
-
-        error_views = error.ErrorViews(exc, pyramid_request)
-        error_views.error()
-
-        assert pyramid_request.raven.captureException.call_count == 1
-
-    def test_it_sets_correct_message_for_missing_lti_param_error_for_missing_lti_launch_params(self, pyramid_request):
-        exc = MissingLTILaunchParamError('test')
-
-        error_views = error.ErrorViews(exc, pyramid_request)
-        resp = error_views.missing_lti_param_error()
+        assert pyramid_request.response.status_int == 400
```

Comment:
Is there a way to test the generic exception case?

Reply:
What do you mean the generic exception case? Tests for the `error()` method are below

Reply:
I meant the unhandled exceptions, tests for @exception_view_config(Exception).

Reply:
Ah I see you have some below now. test_error_does_not_report_exception_to_sentry, etc-looks good!